modules/jflip.cc

```diff
@@ -0,0 +1,291 @@
+ * an input and output port, but doesn't do anything. To customize:
+ *
+ * 1. Replace "modname" with the name of your module.
+ * 2. Add variables for configurable options in modname_options class
+ * 3. Edit modname_options constructor to define commandline options
+ * 4. Edit process() for the realtime logic.
+ * 5. Edit main() for startup and shutdown.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Copyright (C) 2010-2013 C Daniel Meliza <dan || meliza.org>
+ */
+#include <iostream>
+#include <signal.h>
+#include <boost/shared_ptr.hpp>
+
+// #include <sys/time.h>
+#include <stdlib.h>     /* srand, rand */
+#include <time.h>       /* time */
+#include <stdexcept>
+
+#include "jill/logging.hh"
+#include "jill/jack_client.hh"
+#include "jill/program_options.hh"
+
+// https://github.com/lemire/Code-used-on-Daniel-Lemire-s-blog/tree/master/2013/08/14/cpp
+// pick N distinct numbers at random
+// #include "mersenne.h"
+// #include "synthetic.h"
+
+#define PROGRAM_NAME "jflip"
+
+using namespace jill;
+using std::string;
+typedef std::vector<string> stringvec;
+
+class jflip_options : public program_options {
+
+public:
+    jflip_options(string const &program_name);
+
+    /** The server name */
+    string server_name;
+    /** The client name (used in internal JACK representations) */
+    string client_name;
+
+
+    /** A vector of inputs to connect to the client */
+    std::vector<string> input_ports;
+    /** A vector of outputs to connect to the client */
+    std::vector<string> output_ports;
+        /** The MIDI output channel */
+        // midi::data_type output_chan;
+
+    // float buffer_size_s;
+
+
+protected:
+
+    virtual void print_usage();
+
+}; // jflip_options
+
+static jflip_options options(PROGRAM_NAME);
+static boost::shared_ptr<jack_client> client;
+// boost::shared_ptr<dsp::crossing_trigger<sample_t> > trigger;
+// boost::shared_ptr<dsp::buffered_data_writer> arf_thread;
+jack_port_t *port_in, *port_out, *port_trig,*port_count;
+static int ret = EXIT_SUCCESS;
+static int running = 1;
+static int stopping = 0;
+static int xruns = 0;
+
+/* data storage for event times */
+// struct event_t {
+//         nframes_t time;
+//         int status;
+// };
+// dsp::ringbuffer<event_t> trig_times(128);
+
+
+int
+process(jack_client *client, nframes_t nframes, nframes_t)
+{
+    sample_t *in = client->samples(port_in, nframes);
+    sample_t *out = client->samples(port_out, nframes);
+
+    nframes_t inputSize = nframes;
+
+    memcpy(out, in, nframes * sizeof(sample_t));
+    /* rand() method */
+    int i = 0;
+    std::srand(std::time(NULL));
+    while( i < inputSize) { 
+        int coin = rand()%2;
+        if (coin == 0) *(out+i) *= -1;
```

Comment:
This style can be easy to misread for anything more complex than `if (...) break` or `if (...) continue`. Can you bring the multiplication step down onto its own line?